src/map/pc.cpp

```diff
@@ -2089,8 +2089,13 @@ void pc_calc_skilltree(struct map_session_data *sd)
 		uint16 skill_id = skill.second->nameid;
 		uint16 idx = skill_get_index(skill_id);
 
-		if( sd->status.skill[idx].flag != SKILL_FLAG_PLAGIARIZED && sd->status.skill[idx].flag != SKILL_FLAG_PERM_GRANTED ) //Don't touch these
+		if (sd->status.skill[idx].flag != SKILL_FLAG_PLAGIARIZED && sd->status.skill[idx].flag != SKILL_FLAG_PERM_GRANTED) { //Don't touch these
```

Comment:
`sd->state.connect_new != 1 && sd->state.changemap != 1 &&` to prevent temp skills being removed upon connecting and map change.